CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Rodio Development Guide
+
+## Quick Start
+
+1. Install [Rust](https://www.rust-lang.org/tools/install)
+2. Clone the repository: `git clone https://github.com/RustAudio/rodio.git`
+3. Navigate to the project: `cd rodio`
+4. Build the project: `cargo build`
+
+## Project Structure
+
+src/:
+- `source/`: Audio source implementations
+- `decoder/`: Audio format decoders
+- `sink/`: Audio playback and mixing
+- `dynamic_mixer/`: Real-time audio mixing
+- `spatial/`: Spatial audio capabilities
+
+## Coding Guidelines
+
+- Follow [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/)
+- Use `rustfmt` for formatting
+- Implement `Source` trait for new audio sources
+- Use `Sink` for playback management
+
+## Common Tasks
+
+### Adding a New Audio Source or Effect
+
+1. Create a new file in `src/source/`
+2. Implement the `Source` trait to define how audio samples are generated or modified
```

Comment:
lets add a hint about wrapping inner sources. Maybe point to amplify as the easiest example.

Reply:
IDK if u want to do that instead of me, you're probably more qualified for that.

Reply:
I'll get on it :+1: